### ISSUE TITLE
busctl hangs and times out when trying to get a nonexistent property from a nonexistent unit

### ISSUE THREAD

**Issue by Kalsofar** [systemd/systemd#6059]
### Submission type

 - Bug report

### systemd version the issue has been seen with

 v233, v231, v219

*NOTE: Do not submit bug reports about anything but the two most recently released systemd versions upstream!*

### Used distribution

 Fedora 26, Fedora 25, CentOS 7.3

### In case of bug report: Expected behaviour you didn't see

An error message similar to the following:

```
[testuser@fedora-test ~]$ busctl get-property org.freedesktop.systemd1 /org/freedesktop/systemd1/unit/nonexistent org.freedesktop.systemd1.Service nonexistent
Unit name nonexistent is not valid.
```

### In case of bug report: Unexpected behaviour you saw

busctl hangs and times out:

```
[testuser@fedora-test ~]$ busctl get-property org.freedesktop.systemd1 /org/freedesktop/systemd1/unit/nonexistent org.freedesktop.systemd1.Service nonexistent
Connection timed out
```

localectl and systemctl stop working:
```
[testuser@fedora-test ~]$ localectl
Could not get properties: Connection timed out
[testuser@fedora-test ~]$ systemctl
Failed to list units: Connection timed out
```

### In case of bug report: Steps to reproduce the problem

Try getting a nonexistent property from a nonexistent unit:

```
busctl get-property org.freedesktop.systemd1 /org/freedesktop/systemd1/unit/nonexistent org.freedesktop.systemd1.Service nonexistent
```

**Comment by mbiebl** [comment]
Urgh, that's pretty ugly, especially since unprivileged users can run this busctl command. So it's kind of a local DoS. I can confirm that trying to start localed times out, logind seems to get into a confused state as well, when trying to switch to another tty:
```
systemd-logind[436]: Failed to start autovt@tty2.service: Connection timed out
```
Thankfully, systemctl as root still works, as it bypasses dbus-daemon and uses a private socket, so an ordered shutdown is still possible, assuming root was already logged in.
I think this is something which should be fixed as quickly as possible, thus adding to v234 milestone.

**Comment by mbiebl** [comment]
Looks like a ```systemctl daemon-reexec``` is sufficient to get systemd back into a working state.

**Comment by poettering** [comment]
Fix waiting in #6164.
